Jucer2Reprojucer/main.cpp

```diff
@@ -151,6 +151,81 @@ void printError(const juce::String& error)
 }
 
 
+/** Sanitizes a string to match the regex ^[A-Za-z0-9_]*$. Sanitization follows these
+ *  rules:
+ *  1) Invalid characters are replaced with "_".
+ *  2) At most one consecutive replacement "_" is kept (if the original name contained
+ *     multiple "_" they remain unaltered).
+ *  3) No replacement "_" is added at the end or at the beginning.
+ */
+juce::String sanitizeName(const juce::String& input)
+{
+  const juce::String validCharacters =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_";
+  juce::String output;
+  bool mustInsertReplacementChar = false;
+  for (auto c = input.getCharPointer(); !c.isEmpty();)
+  {
+    // keep valid characters
+    if (validCharacters.containsChar(*c))
+    {
+      if (mustInsertReplacementChar)
+      {
+        output += '_';
+        mustInsertReplacementChar = false;
+      }
+      output += *c;
+    }
+    // replace invalid characters with "_"
+    else
+    {
+      // don't add "_" at the beginning
+      if (output.isNotEmpty())
+        mustInsertReplacementChar = true;
+    }
+    c++;
+  }
+  return output;
+}
+
+/** Sanitizes a configuration name. Valid config names match the regex ^[A-Za-z0-9_]*$.
+ *  Sanitization is done with sanitizeName(). An optional digit is added until the
+ *  configuration name is unique. The sanitization is synchronized across
+ *  the entire project.
+ */
+juce::String
+sanitizeConfigurationName(const juce::String& originalName,
+                          std::map<juce::String, juce::String>& sanitizedConfigNameTable)
+{
+  juce::String sanitizedName;
+
+  // check if this name was sanitized before
+  if (sanitizedConfigNameTable.find(originalName) != sanitizedConfigNameTable.end())
+    return sanitizedConfigNameTable[originalName];
+
+  // need to sanitize
+  sanitizedName = sanitizeName(originalName);
+
+  // add digits until unique
+  int number = 1;
+  auto sanitizedNameWithoutNumber = sanitizedName;
+  while (sanitizedConfigNameTable.find(sanitizedName) != sanitizedConfigNameTable.end())
```

Comment:
`sanitizedConfigNameTable.find(sanitizedName)` will always return `sanitizedConfigNameTable.end()` because `sanitizedName` is only used as a value of the map, not as a key.
Using 2 `juce::StringArray` objects, one for the original names and one for the sanitized names, might result in simpler code.